worlds/AutoWorld.py

```diff
@@ -497,22 +495,20 @@ def get_pre_fill_items(self) -> List["Item"]:
 
     # these two methods can be extended for pseudo-items on state
     def collect(self, state: "CollectionState", item: "Item") -> bool:
+        assert item.advancement, "Collect should no longer be called with non-advancements."
+
         """Called when an item is collected in to state. Useful for things such as progressive items or currency."""
         name = self.collect_item(state, item)
-        if name:
-            state.prog_items[self.player][name] += 1
-            return True
-        return False
+        state.prog_items[self.player][name] += 1
+        return True  # Backwards compatibility, this used to denote whether a progression item was picked up
 
     def remove(self, state: "CollectionState", item: "Item") -> bool:
         """Called when an item is removed from to state. Useful for things such as progressive items or currency."""
         name = self.collect_item(state, item, True)
-        if name:
-            state.prog_items[self.player][name] -= 1
-            if state.prog_items[self.player][name] < 1:
-                del (state.prog_items[self.player][name])
-            return True
-        return False
+        state.prog_items[self.player][name] -= 1
+        if state.prog_items[self.player][name] < 1:
+            del (state.prog_items[self.player][name])
+        return True  # Backwards compatibility, this used to denote whether a progression item was picked up
```

Comment:
The only part I'm not happy with - All the overridden world collects are relying on the World collect returning "True" on an advancement, so I can't get rid of `return True` in favor of `return`

If someone has some idea of how to get around this, let me know